File: sportskhan/post_process_sportskhan.py
#!/usr/bin/env python3


import sys
import re
import feedmakerutil


def main():
    urlList = []
    encoding = "cp949"

    for line in feedmakerutil.readStdinAsLineList():
        line = line.rstrip()
        m = re.search(r"<a href='(?P<url>http://[^']+)'[^>]*>\d+</a>", line)
        if m:
            urlList.append(m.group("url"))
        else:
            m = re.search(r"<img src='(?P<url>[^']+)'[^>]*>", line)
            if m:
                print("<img src='%s' width='100%%'/>" % (m.group("url")))

    for url in urlList:
        cmd = "wget.sh '%s' %s" % (url, encoding)
        result = feedmakerutil.execCmd(cmd)
        if result:
            for line in result.split("\n"):
                m = re.search(r"<img\s*[^>]*src=(?:'|\")?(?P<url>http://images.sportskhan.net/article/[^'\"\s]+)(?:'|\")?[^>]*>", line)
                if m:
                    print("<img src='%s' width='100%%'/>" % (m.group("url")))
                                        

if __name__ == "__main__":
    sys.exit(main())

File: _smilekhan/post_process_sportskhan.pl
#!/usr/bin/env perl

use English;
use strict;
use warnings;
use Carp;
use Modern::Perl;
use FeedMaker;


sub main
{
	my @url_list = ();

	while (my $line = <STDIN>) {
		chomp $line;
		if ($line =~ m!<a href='(http://[^']+)'[^>]*>\d+</a>!) {
			push(@url_list, $1);
		} elsif ($line =~ m!<img src='([^']+)'[^>]*>!) {
			print "<img src='" . $1 . "' width='100%'/>\n";
		}
	}

	my $encoding = FeedMaker::getEncodingFromConfig();

	foreach my $url (@url_list) {
		my $cmd = qq(wget.sh "$url");
		#print $cmd . "\n";
		my $result = `$cmd`;
		if ($CHILD_ERROR == 0) {
			for my $line (split /\n/, $result) {
				if ($line =~ m!<img\s*[^>]*src=(?:'|")?(http://images.sportskhan.net/article/[^'"\s]+)(?:'|")?[^>]*>!) {
					print "<img src='" . $1 . "' width='100%'/>\n";
				}
			}
		}
	}
}


main();
